niworkflows/interfaces/morphology.py

```diff
@@ -0,0 +1,105 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# We support and encourage derived works from this project, please read
+# about our expectations at
+#
+#     https://www.nipreps.org/community/licensing/
+#
+""" Handling brain mask"""
+
+from nipype.interfaces.base import (
+    traits,
+    TraitedSpec,
+    BaseInterfaceInputSpec,
+    File,
+    SimpleInterface,
+)
+
+
+class _CrownMaskInputSpec(BaseInterfaceInputSpec):
+    in_segm = File(
+        exists=True, mandatory=True, position=0, desc="Atlas from segmentation."
+    )
+    in_brainmask = File(exists=True, mandatory=True, position=1, desc="Brain mask.")
+    radius = traits.Int(default_value=2, usedefault=True, desc="Radius of dilation")
+
+
+class _CrownMaskOutputSpec(TraitedSpec):
+    out_mask = File(exists=False, desc="Crown mask")
+
+
+class CrownMask(SimpleInterface):
+    """Dilate brain mask for computing the crown mask."""
+
+    input_spec = _CrownMaskInputSpec
+    output_spec = _CrownMaskOutputSpec
+
+    def _run_interface(self, runtime):
+        import nibabel as nb
+        from pathlib import Path
+
+        # Open files
+        segm_img = nb.load(self.inputs.in_segm)
+        brainmask_img = nb.load(self.inputs.in_brainmask)
+
+        segm = segm_img.get_fdata()
+        brainmask = brainmask_img.get_fdata()
+
+        # Obtain dilated brainmask
+        crown_mask, func_seg_mask = get_dilated_brainmask(
+            atlaslabels=segm,
+            brainmask=brainmask,
+            radius=self.inputs.radius,
+        )
+        # Remove the brain from the crown mask
+        crown_mask[func_seg_mask] = False
+        crown_file = str((Path(runtime.cwd) / "crown_mask.nii.gz").absolute())
+        nb.Nifti1Image(
+            crown_mask, brainmask_img.affine, brainmask_img.header
+        ).to_filename(crown_file)
+        self._results["out_mask"] = crown_file
+
+        return runtime
+
+
+def get_dilated_brainmask(atlaslabels, brainmask, radius=2):
+    """Obtain the brain mask dilated
+    Parameters
+    ----------
+    atlaslabels: ndarray
+        A 3D array of integer labels from an atlas, resampled into ``img`` space.
+    brainmask: ndarray
+        A 3D binary array, resampled into ``img`` space.
+    radius: int, optional
+        The radius of the ball-shaped footprint for dilation of the mask.
+    """
+    from scipy import ndimage as ndi
+    from skimage.morphology import ball
+
+    # Binarize the anatomical mask
+    seg_mask = (atlaslabels != 0).astype("uint8")
+
+    # Union of functionally and anatomically extracted masks
+    func_seg_mask = (seg_mask + brainmask) > 0
```

Comment:
Hi @effigies, thanks a lot for your review. atlas_labels is not boolean, that is why I create the binarized mask seg_mask. Thus I propose the following :
```suggestion
def get_dilated_brainmask(atlaslabels, brainmask, radius=2):
    """Obtain the brain mask dilated
    Parameters
    ----------
    atlaslabels: ndarray
        A 3D array of integer labels from an atlas, resampled into ``img`` space.
    brainmask: ndarray
        A 3D binary array, resampled into ``img`` space.
    radius: int, optional
        The radius of the ball-shaped footprint for dilation of the mask.
    """
    from scipy import ndimage as ndi
    from skimage.morphology import ball

    # Binarize the anatomical mask
    seg_mask = (atlaslabels != 0)

    # Union of functionally and anatomically extracted masks
    func_seg_mask = seg_mask | brainmask
```

Reply:
Sorry, see the first suggestion. Coercing the data object with `np.bool(img.dataobj)` should do the same thing.

Reply:
ah ok ! sorry for the misunderstanding. We need however to be careful because I'm using this function in niworkflows/viz/plots.py too (here https://github.com/nipreps/niworkflows/pull/651/files#diff-860ac54523cbe17f5b389cf268694d6050653f6837ce406601b87af4cb1ea95aR269). There atlas_labels is also not binary
PS please let me know if there is a more elegant way to refer to a code line.

Reply:
Suggestions implemented